tests/complex_contracts/nft/erc721.rs

```diff
@@ -0,0 +1,1064 @@
+use std::collections::HashSet;
+
+use cairo_rs::vm::runners::cairo_runner::ExecutionResources;
+use felt::Felt252;
+use num_traits::Zero;
+use starknet_crypto::FieldElement;
+use starknet_rs::business_logic::state::cached_state::CachedState;
+use starknet_rs::{
+    business_logic::{
+        execution::objects::{CallInfo, CallType, OrderedEvent},
+        fact_state::{
+            in_memory_state_reader::InMemoryStateReader, state::ExecutionResourcesManager,
+        },
+        state::state_api::StateReader,
+        transaction::error::TransactionError,
+    },
+    definitions::general_config::StarknetGeneralConfig,
+    services::api::contract_class::EntryPointType,
+    utils::{calculate_sn_keccak, Address},
+};
+
+use crate::complex_contracts::utils::*;
+
+fn contructor(
```

Comment:
please remove these kind of functions since you can just replace it with the call.